packages/connect-form/src/components/advanced-options-tabs/advanced-tab/advanced-tab.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ChangeEvent, useCallback } from 'react';
+import { css } from '@emotion/css';
+import { ConnectionOptions } from 'mongodb-data-service';
+import {
+  RadioBox,
+  RadioBoxGroup,
+  spacing,
+  TextInput,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+import { MongoClientOptions } from 'mongodb';
+
+import FormFieldContainer from '../../form-field-container';
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+import { ConnectionFormError } from '../../../utils/connect-form-errors';
+import { readPreferences } from '../../../utils/read-preferences';
+
+import UrlOptions from './url-options';
+
+const containerStyles = css({
+  marginTop: spacing[3],
+});
+
+const fieldStyles = css({
+  width: '50%',
+});
+
+function AdvancedTab({
+  updateConnectionFormField,
+  connectionStringUrl,
+}: {
+  errors: ConnectionFormError[];
+  connectionStringUrl: ConnectionStringUrl;
+  hideError: (errorIndex: number) => void;
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions?: ConnectionOptions;
+}): React.ReactElement {
+  const handleFieldChanged = useCallback(
+    (key: keyof MongoClientOptions, value: unknown) => {
+      updateConnectionFormField({
+        type: 'update-search-param',
+        key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  const readPreference = connectionStringUrl.searchParams.get('readPreference');
+  const replicaSet = connectionStringUrl.searchParams.get('replicaSet');
+  const authSource = connectionStringUrl.searchParams.get('authSource');
+
+  return (
+    <div className={containerStyles}>
+      {/* Read Preferences */}
+      <RadioBoxGroup
+        onChange={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+          handleFieldChanged('readPreference', value);
+        }}
+        data-testid="read-preferences"
+      >
+        {readPreferences.map(({ title, id }) => {
+          return (
+            <RadioBox
+              data-testid={`${id}-preference-button`}
+              checked={readPreference === id}
+              value={id}
+              key={id}
+            >
+              {title}
+            </RadioBox>
+          );
+        })}
+      </RadioBoxGroup>
+      {/* Replica Set */}
+      <FormFieldContainer>
+        <TextInput
+          className={fieldStyles}
+          onChange={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            handleFieldChanged('replicaSet', value);
+          }}
+          name={'replica-set'}
+          data-testid={'replica-set'}
+          label={'Replica Set Name'}
+          type={'text'}
+          optional={true}
```

Comment:
We might want to do some custom styles to make the optional vertically centered like how it looks in lg:
<img width="388" alt="Screen Shot 2022-01-05 at 1 30 44 AM" src="https://user-images.githubusercontent.com/1791149/148179449-1d71740f-11ad-48fe-b437-ed4659e07f3a.png">
vs currently:
<img width="431" alt="Screen Shot 2022-01-05 at 1 30 48 AM" src="https://user-images.githubusercontent.com/1791149/148179473-b454d2ae-0d27-4590-a37b-a5c0bf0f296a.png">
Definitely fine how it is too.

Reply:
So the global styles that are breaking this leafygreen style are coming from this import:

https://github.com/mongodb-js/compass/blob/ab650e0c7926da8df9905ce474114a7b9bc649e0/packages/compass/src/app/styles/bootstrap.less#L15

I'd suggest we try to remove this import and see if anything else breaks, if it's not that bad I think we should try to remove as much of those global styles as possible as they will always create problems for us

Reply:
this certainly resolves the issue but I am not sure of the implications. I'll check this out with @Anemy

Reply:
Not sure of the implications either, or why we have it. I'd say go for it but click around Compass a bit after and see if any styles seem different, I'm happy to also test about on this with you. Might be that we can remove more of these bootstrap styles.